packages/backend-auth/src/userpool_access_policy_factory.ts

```diff
@@ -22,127 +14,99 @@ export class UserPoolAccessPolicyFactory {
+    const policyActions: Set<string> = new Set();
+
+    actions.forEach((authAction) => {
+      const mappedAction = iamActionMap[authAction];
+
+      if (typeof mappedAction === 'string') {
+        policyActions.add(mappedAction);
+      } else {
+        mappedAction.forEach((action) => policyActions.add(action));
+      }
+    });
+
+    if (policyActions.size === 0) {
       throw new AmplifyFault('EmptyPolicyFault', {
         message: 'At least one action must be specified',
       });
     }
 
-    const statements: PolicyStatement[] = [];
-
-    actions.forEach((action) => {
-      statements.push(this.getStatement(action));
-    });
+    const policy = new Policy(
+      this.stack,
+      `${this.namePrefix}${this.policyCount++}`,
+      {
+        statements: [
+          new PolicyStatement({
+            actions: [...policyActions],
+            resources: [this.userpool.userPoolArn],
+          }),
+        ],
+      }
+    );
 
-    return new Policy(this.stack, `${this.namePrefix}${this.policyCount++}`, {
-      statements,
-    });
+    return policy;
   };
-
-  private getStatement = (action: UserpoolAction) =>
-    new PolicyStatement({
-      actions: actionPolicyMap[action],
-      resources: [this.userpool.userPoolArn],
-    });
 }
 
-const actionPolicyMap: Record<UserpoolAction, string[]> = {
-  create: [
-    'cognito-idp:ConfirmSignUp',
-    'cognito-idp:AdminCreateUser',
-    'cognito-idp:CreateUserImportJob',
-    'cognito-idp:AdminSetUserSettings',
-    'cognito-idp:AdminLinkProviderForUser',
-    'cognito-idp:CreateIdentityProvider',
+type IamActionMap = {
+  [action in ActionIam]: string;
+} & {
+  [amplifyActionSet in ActionMeta]: string[];
+};
+
+const iamActionMap: IamActionMap = {
+  manageUser: [
+    'cognito-idp:AdminAddUserToGroup',
     'cognito-idp:AdminConfirmSignUp',
+    'cognito-idp:AdminCreateUser',
+    'cognito-idp:AdminDeleteUser',
+    'cognito-idp:AdminDeleteUserAttributes',
     'cognito-idp:AdminDisableUser',
-    'cognito-idp:AdminRemoveUserFromGroup',
-    'cognito-idp:SetUserMFAPreference',
-    'cognito-idp:SetUICustomization',
-    'cognito-idp:SignUp',
-    'cognito-idp:VerifyUserAttribute',
-    'cognito-idp:SetRiskConfiguration',
-    'cognito-idp:StartUserImportJob',
-    'cognito-idp:AdminSetUserPassword',
-    'cognito-idp:AssociateSoftwareToken',
-    'cognito-idp:CreateResourceServer',
-    'cognito-idp:RespondToAuthChallenge',
-    'cognito-idp:CreateUserPoolClient',
-    'cognito-idp:AdminUserGlobalSignOut',
-    'cognito-idp:GlobalSignOut',
-    'cognito-idp:AddCustomAttributes',
-    'cognito-idp:CreateGroup',
-    'cognito-idp:CreateUserPool',
+    'cognito-idp:AdminEnableUser',
     'cognito-idp:AdminForgetDevice',
-    'cognito-idp:AdminAddUserToGroup',
+    'cognito-idp:AdminGetDevice',
+    'cognito-idp:AdminGetUser',
+    'cognito-idp:AdminListDevices',
+    'cognito-idp:AdminListGroupsForUser',
+    'cognito-idp:AdminListUserAuthEvents',
```

Comment:
Reading the docs for this action I don't think it belongs in user management: https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_AdminListUserAuthEvents.html

it looks more like a "userAudit" action